pwndbg/stdio.py

```diff
@@ -20,28 +20,18 @@
 
 
 def get(fd, mode):
-    if pwndbg.compat.python3:
-        file = io.open(fd, mode=mode, buffering=0)
-
-        kw ={}
-        if pwndbg.compat.python3:
-            kw['write_through']=True
-
-        return io.TextIOWrapper(file, **kw)
-    else:
-        return codecs.open(fd, mode, 'utf-8')
+    return io.open(fd, mode + 'b', buffering=0)
 
 
 class Stdio(object):
     queue = []
 
     def __enter__(self, *a, **kw):
         self.queue.append((sys.stdin, sys.stdout, sys.stderr))
-
         if pwndbg.compat.python3 or True:
```

Comment:
Yes, it's some left-over stuff from troubleshooting this code.

tl;dr is that having a non-buffering file descriptor which transparently takes any of `str`, `unicode` (Python2), and `bytes` (Python3) is annoying.  I think I found a least-common denominator that works, though.